hicommonwealth/commonwealth
# User-Onboarding: Join community step

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6648

## Description of Changes
Added the 3rd step -> Join community step in the welcome onboard modal, which would suggest communities that match the auth user's chain network and allow to join those form the suggested list.

## "How We Fixed It"
N/A

## Test Plan
- Create a new account (this can be any account type)
- Login
- Apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts b/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
index 34e0f716ec..bf99934991 100644
--- a/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
+++ b/packages/commonwealth/client/scripts/state/ui/modals/welcomeOnboardModal.ts
@@ -9,7 +9,7 @@ interface WelcomeOnboardModalProps {
 
 export const welcomeOnboardModal = createStore<WelcomeOnboardModalProps>()(
   devtools((set) => ({
-    isWelcomeOnboardModalOpen: false,
+    isWelcomeOnboardModalOpen: true,
     setIsWelcomeOnboardModalOpen: (isOpen) => {
       set((state) => {
         return {
diff --git a/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx b/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
index c3601ea350..cfb8dd6d7f 100644
--- a/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
+++ b/packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/WelcomeOnboardModal.tsx
@@ -10,7 +10,7 @@ import { WelcomeOnboardModalProps, WelcomeOnboardModalSteps } from './types';
 
 const WelcomeOnboardModal = ({ isOpen, onClose }: WelcomeOnboardModalProps) => {
   const [activeStep, setActiveStep] = useState<WelcomeOnboardModalSteps>(
-    WelcomeOnboardModalSteps.PersonalInformation,
+    WelcomeOnboardModalSteps.JoinCommunity,
   );
```
- Verify you see the 3rd step of welcome modal (aka Join Communities)
- Verify that once you join a community, its `Join community` button changes to `Joined` and is not clickable anymore
- Verify that you can join all communities (for now we only suggest 4 communities)
- Verify that you can skip this step by clicking on `Let's go` or closing the modal.

## Deployment Plan
N/A

## Other Considerations
N/A
